synapse/federation/federation_server.py

```diff
@@ -911,6 +919,10 @@ async def on_edu(self, edu_type: str, origin: str, content: dict):
         if not self.config.use_presence and edu_type == "m.presence":
             return
 
+        # If the incoming EDUs from this origin/type are over the limit, drop them.
+        if not self._edu_rate_limiter.can_do_action(key=(edu_type, origin)):
+            return
```

Comment:
Does this make sense to limit all types of EDUs or should we be limiting it to `m.room_key_request` as in #8675?

Reply:
This scares me a lot TBH, especially since this includes typing/read receipts/presence etc. I imagine e.g. matrix.org sends quite a lot of those.